Codegen/test/Performance.cs

```diff
@@ -0,0 +1,232 @@
+                Path.GetDirectoryName(path),
+                new[] { ".tcbl", ".cbl", ".cpy" },
+                DocumentFormat.RDZReferenceFormat,
+                new TypeCobolOptions());
+            FileCompiler fileCompiler = new FileCompiler(compilationProject, fileName, false);
+            fileCompiler.CompileOnce();
+
+            // Generator.
+            StringBuilder destination = new StringBuilder();
+            var columnsLayout = fileCompiler.CompilationResultsForProgram.ProgramClassDocumentSnapshot.TextSourceInfo.ColumnsLayout;
+            IGenerator generator = new DefaultGenerator(fileCompiler.CompilationResultsForProgram, destination, null, null);
+
+            // Codegen iterations.
+            Console.WriteLine("FILE;ITERATION;STEP;TIME_TAKEN");
+            Dictionary<string, TimeSpan> total = new Dictionary<string, TimeSpan>();
+            for (int i = 0; i < ITERATION_COUNT; i++)
+            {
+                // Clear target StringBuilder and force recompile.
+                destination.Clear();
+                if (forceFullParsing)
+                {
+                    // Full Compile.
+                    fileCompiler = new FileCompiler(compilationProject, fileName, false);
+                    fileCompiler.CompileOnce();
+                }
+                else
+                {
+                    // Incremental compile.
+                    ITextLine newLine = new TextLineSnapshot(fileCompiler.CompilationResultsForProgram.CobolTextLines.Count, string.Empty, null);
+                    TextChangedEvent textChangedEvent = new TextChangedEvent();
+                    textChangedEvent.TextChanges.Add(new TextChange(TextChangeType.LineInserted, newLine.LineIndex, newLine));
+                    fileCompiler.CompilationResultsForProgram.UpdateTextLines(textChangedEvent);
+                    fileCompiler.CompileOnce();
+                }
+
+                // Generate and write stats.
+                generator.Generate(fileCompiler.CompilationResultsForProgram, columnsLayout);
+                foreach (var pair in generator.PerformanceReport)
+                {
+                    Console.WriteLine($"{fileName};{i+1};{pair.Key};{pair.Value.TotalMilliseconds}");
+                    if (!total.ContainsKey(pair.Key))
+                    {
+                        total.Add(pair.Key, TimeSpan.Zero);
+                    }
+                    total[pair.Key] += pair.Value;
+                }
+            }
+
+            // Aggregated stats.
+            Console.WriteLine();
+            Console.WriteLine("Time per step :");
+            foreach (var stepData in total.Select((p, i) => new {Index = i + 1, Step = p.Key, Duration = p.Value.TotalMilliseconds}).OrderByDescending(a => a.Duration))
+            {
+                Console.WriteLine($"{stepData.Step,-25}({stepData.Index}) : Total = {stepData.Duration,15} ms, Avg = {stepData.Duration / ITERATION_COUNT,15} ms");
```

Comment:
Why -25 and 15?

Reply:
To slightly improve readability of results. -25 left aligns the step name on 25 characters and 15 right aligns the duration. It assumes that the step name is 25 characters long max.